dev/fattest.simplicity/src/componenttest/rules/repeater/RepeatFeatureSet.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package componenttest.rules.repeater;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RepeatFeatureSet {
+    private final String id;
+    private final Set<String> features;
+
+    public RepeatFeatureSet(String id, Set<String> features) {
+        this.id = id;
+        this.features = Collections.unmodifiableSet(features);
+    }
+
+    public String getID() {
+        return this.id;
+    }
+
+    public RepeatFeatureSet withFeature(String id, String feature) {
+        Set<String> newFeatures = new HashSet<String>(this.features);
+        newFeatures.add(feature);
+        RepeatFeatureSet newSet = new RepeatFeatureSet(id, newFeatures);
+        return newSet;
+    }
+
+    public RepeatFeatureSet removeFeature(String id, String feature) {
+        Set<String> newFeatures = new HashSet<String>(this.features);
+        newFeatures.remove(feature);
+        RepeatFeatureSet newSet = new RepeatFeatureSet(id, newFeatures);
+        return newSet;
+    }
+
+    public RepeatFeatureSet withFeature(String feature) {
+        return withFeature(this.id, feature);
+    }
+
+    public RepeatFeatureSet removeFeature(String feature) {
+        return removeFeature(this.id, feature);
+    }
+
+    public RepeatFeatureSet withID(String id) {
+        return new RepeatFeatureSet(id, this.features);
+    }
```

Comment:
I worry a little that this will confuse people, since it looks similar to the methods on `FeatureReplacementAction` but it works differently.

I do like the ability to take an modify an existing feature set.

I don't like that you can add features to an existing feature set without changing the ID.

I think it would be better if the ID had to be provided in the constructor, but you could copy the feature list from another set. E.g:
```
new RepeatFeatureSet(ID).addAllFrom(MP40).remove("mpConfig-2.0").add("mpConfig-2.0");
```

However, this wouldn't stop someone doing this and getting a featureset with the MP4.0 ID.
```
MP40.add("someOtherFeature");
```

Not sure if you can prevent that without using a builder pattern.

Reply:
went with a partial builder pattern